frontend/src/screens/AdminEventsPanel.tsx

```diff
@@ -24,53 +24,92 @@ export interface Event {
 
 function AdminEventsPanel() {
   const [ranksGroups, setRanksGroups] = useState<string[]>();
+  const [numberOfRanks, setNumberOfRanks] = useState<number>(0);
+  const [ranksAddress, setRanksAddress] = useState<string | null>(null);
 
   const { register, handleSubmit } = useForm<FormInputs>();
-  const { fetchRanksNames, getRanksContractAddress } = useEthereum();
-
-  const createEvent = async (event: Event) => {
-    const ranksContractAddress = await getRanksContractAddress(
-      event.ranksAddress
-    );
-
-    console.log(ranksContractAddress);
-
-    const ranksContract = createContract(
-      DynamicContracts.RANKS,
-      ranksContractAddress
-    );
-
-    console.log(ranksContract);
-  };
+  const {
+    fetchRanksNames,
+    getRanksContractAddress,
+    getRanksNumber,
+    createEvent,
+  } = useEthereum();
+
+  const showRanksForm = handleSubmit(async ({ ranksGroup }) => {
+    const rankAddress = await getRanksContractAddress(ranksGroup);
+    setRanksAddress(rankAddress);
+    const ranksNumber = await getRanksNumber(rankAddress);
+    setNumberOfRanks(+ranksNumber);
+  });
 
   useEffect(() => {
     fetchRanksNames().then((names) => setRanksGroups(names));
   }, []);
 
-  const onSubmit = handleSubmit(
-    async ({
-      eventName,
-      eventDescription,
-      eventDate,
-      eventTicketsNumber,
-      ranksGroup,
-    }) => {
-      const event: Event = {
-        ranksAddress: "mock", /// TODO: ADD PROPER PARSING!!!
-        name: "mock",
-        symbol: "mock",
-        saleStartTimePerRank: [],
-        maxTicketsPerUserPerRank: [],
-        ticketPricePerRank: [],
-      };
-
-      createEvent(event);
-    }
-  );
+  const onSubmit = handleSubmit(async (values) => {
+    const saleStartTimePerRank: number[] = [];
+    const maxTicketsPerUserPerRank: number[] = [];
+    const ticketPricePerRank: number[] = [];
+
+    Object.keys(values).forEach((key) => {
+      const index = +key.match(/\d+/)!;
+      //@ts-ignore
+      const value = values[key];
+
+      if (key.includes("saleStartTime")) {
+        saleStartTimePerRank[index] = new Date(value).getTime();
+      } else if (key.includes("maxTicketsPerUser")) {
+        maxTicketsPerUserPerRank[index] = +value;
+      } else if (key.includes("ticketPrice")) {
+        ticketPricePerRank[index] = value;
+      }
+    });
+
+    const event: CreateEventRequest = {
+      ticketName: values.name,
+      ticketSymbol: values.symbol,
+      ranksAddress: ranksAddress!, // I shouldnt do it, but ...
```

Comment:
assuming that the rank group exists and it is required in the form I think that `const [ranksAddress, setRanksAddress] = useState("");` should work fine. numberOfRanks hides submit button so suggested approach should be safe